src/Fmt_ast.ml

```diff
@@ -383,7 +383,10 @@ let fmt_let c ctx ~ext ~rec_flag ~bindings ~body ~parens ~attributes
                   ?ext:(if first then ext else None)
                   ctx binding
                   ~in_:(fun indent ->
-                    fmt_if_k last (break 1 (-indent) $ fmt "in") )
+                    fmt_if_k last
+                      (fmt_or_k c.conf.break_before_in
+                         (break 1 (-indent) $ fmt "in")
+                         (fits_breaks " in" "@;<1 -2>in")) )
```

Comment:
Could you try using `Format.pp_print_fits_or_breaks` directly (or through a new function in `Fmt`) to avoid hardcoding the indent ?

Reply:
Good catch! Thanks!

I fixed it but I went for an easier solution: build the required string using `indent`. I think the two commits should be squashed when merged.

I can add a function to the `Fmt` module if it's more in line with the usage of the project. My preferred solution would be something along the lines of `fits_or: string -> t -> t` which checks if there's enough space for the `string` and otherwise uses `t`. I can't see the necessary helper functions in `Format` though.

Reply:
It's better but I'm not fan of the approach.
`fits_breaks` cannot take a format as parameter, because it is based on `Format.pp_print_fits_or_breaks` that only takes strings. The "@;\<n o>" part is actually parsed here: https://github.com/ocaml-ppx/ocamlformat/blob/master/src/Fmt.ml#L103
This is obfuscation to me and I would prefer a direct binding to `pp_print_fits_or_breaks` but I understand this may be out of the scope of this PR.